File: FE/src/app/modules/shared/model/document-generator.ts
export class DocumentGenerator {
  private _id: number;
  private _name: string;
  private _content: string;
  private _type: string;
  private _active: boolean;


  public get id(): number {
    return this._id;
  }
  public set id(id: number) {
    this._id = id;
  }

  public get name(): string {
    return this._name;
  }
  public set name(name: string) {
    this._name = name;
  }

  public get content(): string {
    return this._content;
  }
  public set content(content: string) {
    this._content = content;
  }

  public get type(): string {
    return this._type;
  }
  public set type(type: string) {
    this._type = type;
  }

  public get active(): boolean {
    return this._active;
  }
  public set active(active: boolean) {
    this._active = active;
  }

  public get toJson() {
    return {
      id: this.id ? this.id : null,
      name: this.name,
      content: this.content,
      type: this.type,
      active: this.active,
    };
  }

  public fromJson(jsonData: any) {
    this.id = jsonData.id;
    this.name = jsonData.name;
    this.content = jsonData.content;
    this.type = jsonData.type;
    this.active = jsonData.active;
  }
}

File: BE/src/main/java/com/application/core/shared/entities/DocumentGenerator.java

package com.application.core.shared.entities;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class DocumentGenerator  {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    private String name;
     @Column(columnDefinition ="LONGTEXT")
    private String content;
    private String type;

    @Column(columnDefinition = "tinyint(1) default 1")
    private Boolean active = true;



    public JsonNode toJson() {
        ObjectMapper objectMapper = new ObjectMapper();
        JsonNode jsonNode = objectMapper.valueToTree(this);
        return jsonNode;
    }
}
